conversation/api.go

```diff
@@ -21,205 +18,48 @@ const (
 	// path is the path for the Conversation resource, relative to apiRoot.
 	path = "conversations"
 
+	// startConversationPath is the path for starting new conversation
+	startConversationPath = "start"
+
+	// contactPath is the path for fetching a collection of conversations by contact ID
+	contactPath = "contact"
+
 	// messagesPath is the path for the Message resource, relative to apiRoot
 	// and path.
 	messagesPath = "messages"
 
+	// sendMessagePath is the path for creating the Message resource relative to apiRoot
+	sendMessagePath = "send"
+
 	// webhooksPath is the path for the Webhook resource, relative to apiRoot.
 	webhooksPath = "webhooks"
 )
 
-type ConversationList struct {
-	Offset     int
-	Limit      int
-	Count      int
-	TotalCount int
-	Items      []*Conversation
-}
-
-type Conversation struct {
-	ID                   string
-	ContactID            string
-	Contact              *Contact
-	LastUsedChannelID    string
-	Channels             []*Channel
-	Messages             *MessagesCount
-	Status               ConversationStatus
-	CreatedDatetime      time.Time
-	UpdatedDatetime      *time.Time
-	LastReceivedDatetime *time.Time
-}
-
-type Contact struct {
-	ID            string
-	Href          string
-	MSISDN        string
-	FirstName     string
-	LastName      string
-	CustomDetails map[string]interface{}
-	CreatedAt     *time.Time
-	UpdatedAt     *time.Time
-}
-
-type Channel struct {
-	ID              string
-	Name            string
-	PlatformID      string
-	Status          string
-	CreatedDatetime *time.Time
-	UpdatedDatetime *time.Time
-}
-
-type MessagesCount struct {
-	HRef       string
-	TotalCount int
-}
-
-// ConversationStatus indicates what state a Conversation is in.
-type ConversationStatus string
-
-const (
-	// ConversationStatusActive is returned when the Conversation is active.
-	// Only one active conversation can ever exist for a given contact.
-	ConversationStatusActive ConversationStatus = "active"
-
-	// ConversationStatusArchived is returned when the Conversation is
-	// archived. When this is the case, a new Conversation is created when a
-	// message is received from a contact.
-	ConversationStatusArchived ConversationStatus = "archived"
-)
-
-type MessageList struct {
-	Offset     int
-	Limit      int
-	Count      int
-	TotalCount int
-	Items      []*Message
+type GetRequest interface {
```

Comment:
It is attempt to make one standard approach forming query params from params structures instead of many different but similar functions. Actually this interface doesn't use as function param, but I add it as template to all requests which supposed to be query params.